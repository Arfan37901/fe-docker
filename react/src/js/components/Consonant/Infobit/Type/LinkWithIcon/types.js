import {
    bool,
    string,
    oneOfType,
} from 'prop-types';

// eslint-disable-next-line import/prefer-default-export
export const LinkWithIconType = {
    src: string,
    href: string,
    text: string,
    linkHint: string,
    srcAltText: string,
    openInNewTab: oneOfType([string, bool]),
};
