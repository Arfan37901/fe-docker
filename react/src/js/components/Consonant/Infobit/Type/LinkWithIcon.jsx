import React from 'react';
import PropTypes from 'prop-types';

function LinkWithIcon({
    href, openInNewTab, linkHint, text, src, srcAltText,
}) {
    return (
        <a
            href={href}
            target={openInNewTab ? '__blank' : '_self'}
            className="consonant-link-widh-ico-infobit"
            title={linkHint}
            rel="noreferrer"
            tabIndex="0">
            {src && <img src={src} alt={srcAltText} loading="lazy" height="18" />}
            <span>{text}</span>
        </a>
    );
}

LinkWithIcon.propTypes = {
    href: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    src: PropTypes.string,
    openInNewTab: PropTypes.bool,
    linkHint: PropTypes.string,
    srcAltText: PropTypes.string,
};

LinkWithIcon.defaultProps = {
    src: '',
    openInNewTab: true,
    linkHint: '',
    srcAltText: '',
};

export default LinkWithIcon;
