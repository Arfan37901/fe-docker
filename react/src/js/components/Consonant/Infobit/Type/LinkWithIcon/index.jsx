import React from 'react';

import { If } from '../../../Common';
import { LinkWithIconType } from './types';

const defaultProps = {
    src: '',
    href: '',
    text: '',
    linkHint: '',
    srcAltText: '',
    openInNewTab: true,
};

/**
 * LinkWithIcon Infobit (shown in 3:2 Card Footer)
 *
 * @component
 * @example
 * const props= {
    href: String,
    openInNewTab: Boolean,
    linkHint: String,
    text: String,
    src: String,
    srcAltText: String,
 * }
 * return (
 *   <LinkWithIcon {...props}/>
 * )
 */
const LinkWithIcon = ({
    href,
    openInNewTab,
    linkHint,
    text,
    src,
    srcAltText,
}) => (
    <a
        href={href}
        data-testid="link-with-icon"
        target={openInNewTab ? '_blank' : '_self'}
        className="consonant-link-with-ico-infobit"
        title={linkHint}
        rel="noopener noreferrer"
        tabIndex="0">
        <If condition={Boolean(src)}>
            <img src={src} alt={srcAltText} loading="lazy" height="18" />
        </If>
        <span>{text}</span>
    </a>
);

LinkWithIcon.propTypes = LinkWithIconType;
LinkWithIcon.defaultProps = defaultProps;

export default LinkWithIcon;
