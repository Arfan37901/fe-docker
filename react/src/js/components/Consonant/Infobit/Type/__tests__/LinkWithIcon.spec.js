import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import LinkWithIcon from '../LinkWithIcon';

const props = {
    href: '/some/valid/uri',
    openInNewTab: false,
    src: '/some/valid/uri',
    stcAltText: 'text',
    text: 'text',
};

describe('Consonant/Bookmarks', () => {
    test('should correct render', async () => {
        const tree = renderer
            .create(<LinkWithIcon {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('should correct render with icon', async () => {
        const tree = renderer
            .create(<LinkWithIcon {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
