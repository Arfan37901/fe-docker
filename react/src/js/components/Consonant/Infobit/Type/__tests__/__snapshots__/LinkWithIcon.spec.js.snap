// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Bookmarks should correct render 1`] = `
<a
  className="consonant-link-with-ico-infobit"
  href="/some/valid/uri"
  rel="noreferrer"
  tabIndex="0"
  target="_self"
  title=""
>
  <img
    alt=""
    height="18"
    loading="lazy"
    src="/some/valid/uri"
  />
  <span>
    text
  </span>
</a>
`;

exports[`Consonant/Bookmarks should correct render with icon 1`] = `
<a
  className="consonant-link-with-ico-infobit"
  href="/some/valid/uri"
  rel="noreferrer"
  tabIndex="0"
  target="_self"
  title=""
>
  <img
    alt=""
    height="18"
    loading="lazy"
    src="/some/valid/uri"
  />
  <span>
    text
  </span>
</a>
`;
